### PR TITLE
feat: add soil moisture and freeze thaw init config models

### PR DESCRIPTION
**Pull request by aaraney**
## Additions

- Pydantic models for [`SoilFreezeThaw`](https://github.com/noaa-owp/soilFreezeThaw), [`SoilMoistureProfile`](https://github.com/noaa-owp/soilMoistureProfiles/) and [`LGAR-C`](https://github.com/NOAA-OWP/LGAR-C) configuration files

## Notes
- Depends on #95
- Depends on #104

### CODE REVIEW

**Comment by hellkite500** [comment]
Were these TODO's ever addressed?
> satpsi defaults to 0.3
> check initial_psi lower bound
And can you rebase the conflict?

Active development on lgar BMI may require some iterative changes to the lgar init config, but I think we can put these in place and track updates/issues as they are used

**Comment by aaraney** [comment]
> Were these TODO's ever addressed?
> 
> > satpsi defaults to 0.3
> > check initial_psi lower bound

Nope and looking back I don't think there was enough information in the source code of the bmi modules for me to gather that information. So, the TODOs were more long term, "hey, im thinking about this long term." IMO, I don't think they need to be addressed in this PR.

**Comment by aaraney** [comment]
@hellkite500, just rebased and looked through things again. This should be good to merge now!

File: python/ngen_conf/src/ngen/config/init_config/soil_freeze_thaw.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+m = Literal["m"]
+m_per_m = Literal["m/m"]
+k = Literal["K"]
+empty = Literal[""]
+# empty string is equivalent to h / hr
+time_unit = Literal["s", "sec", "", "h", "hr", "d", "day"]
+
+
+class IceFractionScheme(str, Enum):
```
Comment (hellkite500) [inline python/ngen_conf/src/ngen/config/init_config/soil_freeze_thaw.py:19]: I thought these were rainfall runoff partitioning schemes, not Ice Fraction schemes 🤯

**Review by hellkite500 — APPROVED** [review]

**Pull request merged**